### PR TITLE
Python3 + pylint + pep8

### PR DESCRIPTION
**Pull request by Laurits7**
This pull request includes:
- Python2 - Python3 compatibility
- pep8

Regarding pep8, not every line that was too long was broke down simply due to easier readability.

### CODE REVIEW

**Comment by gouskos** [comment]
When I try to run: python/postprocessing/examples/exampleAnalysis.py, I get the following error:
Traceback (most recent call last):
  File "exampleAnalysis.py", line 45, in <module>
    p.run()
  File "/afs/cern.ch/work/g/gouskos/private/xpog/CMSSW_11_2_0_pre7/python/PhysicsTools/NanoAODTools/postprocessing/framework/postprocessor.py", line 169, in run
    print('Pre-select %d entries out of %s (%.2f%%)' % (elist.GetN() if elist else nEntries, nEntries, old_div((elist.GetN() if elist else nEntries), (0.01 * nEntries)) if nEntries else 0))
NameError: global name 'old_div' is not defined

Can you please check?

**Comment by Laurits7** [comment]
No physics validation was done, though I am willing to do it if somebody could point me what to check.

**Comment by gouskos** [comment]
OK. Let me reach JME conveners and I will come back to you.
[We are working on including a validation set in the workflow]

**Comment by camclean** [comment]
@Laurits7 can you add a description to this PR, including a summary of the purpose of all of the file changes? I'm not quite sure why so many files were changed.

File: python/postprocessing/framework/postprocessor.py [hunk 1]
```diff
@@ -1,171 +1,198 @@
...
+            if inTree == None:
+                inTree = inFile.Get("Friends")
+            nEntries = min(inTree.GetEntries() -
+                           self.firstEntry, self.maxEntries)
+            totEntriesRead += nEntries
+            # pre-skimming
+            elist, jsonFilter = preSkim(
+                inTree, self.json, self.cut, maxEntries=self.maxEntries, firstEntry=self.firstEntry)
+            if self.justcount:
+                print('Would select %d / %d entries from %s (%.2f%%)' % (elist.GetN() if elist else nEntries, nEntries, fname, o(elist.GetN() if elist else nEntries) / (0.01 * nEntries) if nEntries else 0))
```
Comment (hqucms) [inline python/postprocessing/framework/postprocessor.py:163]:
```suggestion
                print('Would select %d / %d entries from %s (%.2f%%)' % (elist.GetN() if elist else nEntries, nEntries, fname, (elist.GetN() if elist else nEntries) / (0.01 * nEntries) if nEntries else 0))
```

**Comment by Laurits7** [comment]
I made the event-by-event comparison of the  100 ttbar events postprocessed with the current 'master' branch and the PR (Python3 + pylint + pep8 #251) and found no discrepancies. The .root file of the differences can be found in the following folder (the histograms from the .root file are gathered to combined.pdf in the same folder):
/afs/cern.ch/user/l/ltani/public/py2_to_py3

**Comment by gouskos** [comment]
Thank you @Laurits7 !

**Comment by Laurits7** [comment]
Just to follow up, is it ready to be merged?

File: python/postprocessing/framework/crabhelper.py [hunk 2]
```diff
@@ -5,62 +5,63 @@
...
-            crabFiles[i]=pfn
-            testfile.Close()
-            #tested=True
-         else :
-            print "Test open failed, forcing AAA"
-            crabFiles[i]="root://cms-xrd-global.cern.ch/"+crabFiles[i]
-            forceaaa=True
-       else :
-            crabFiles[i]=pfn
+    print("ARGV: " + sys.argv)
```
Comment (andreypz) [inline python/postprocessing/framework/crabhelper.py:10]: This does not work in python 2

**Review by andreypz — COMMENTED** [review]
After these changes is the code supposed to work in python 2 (deafalt in CMSSW_11_0_0)?

File: python/postprocessing/framework/crabhelper.py (hunk 2 above)
Comment (Laurits7) [reply python/postprocessing/framework/crabhelper.py:10]: Sorry about that, indeed str() is missing.

**Comment by andreypz** [comment]
Was the runtime checked for this merge?
It seems to run significantly slower then before...

**Pull request merged**